File: sdk/java/src/main/java/com/lightswitch/domain/Keyword.java
package com.lightswitch.domain;

import java.util.List;

public class Keyword {
	private List<Property> properties;
	private String value;

	public Keyword(List<Property> properties, String value) {
		this.properties = properties;
		this.value = value;
	}

	public List<Property> getProperties() {
		return properties;
	}

	public String getValue() {
		return value;
	}
}

File: backend/core-service/src/main/kotlin/com/lightswitch/core/domain/flag/repository/entity/Keyword.kt
package com.lightswitch.core.domain.flag.repository.entity

import com.lightswitch.core.common.entity.BaseEntity
import com.lightswitch.core.domain.history.dto.PreKeyword
import jakarta.persistence.*

@Entity(name = "keyword")
class
Keyword(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val keywordId: Long? = null,

    @JoinColumn(name = "flag_id")
    @ManyToOne(fetch = FetchType.LAZY)
    val flag: Flag,

    @OneToMany(mappedBy = "keyword", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    var properties: MutableList<Property> = mutableListOf(),

    var description: String,
    var value: String,
) : BaseEntity() {

    fun toPrevious(): PreKeyword {
        return PreKeyword(this)
    }
}